search/search-domain/src/main/java/io/camunda/search/sort/GroupSort.java

```diff
@@ -25,7 +25,7 @@ public static final class Builder extends AbstractBuilder<Builder>
       implements ObjectBuilder<GroupSort> {
 
     public Builder groupKey() {
-      currentOrdering = new FieldSorting("key", null);
+      currentOrdering = new FieldSorting("groupKey", null);
```

Comment:
❌ With this change the sorting has a different name as the actual property in the GroupEntity and GroupItem. This will be confusing for the end user and they should be the same.

💭 This actually applies to the GroupFilter object as well. Probably we should align all classes/dtos to "groupKey"

This renaming also is the reason for the failing RDBMS tests. We map the property names in the GroupSearchColumn enum to actual DB column names.

Reply:
Hey @stoerti, I was considering aligning the `GroupEntity` classes as well, but I noticed they are used heavily in the RDBMS code, and didn't want to cause merge conflicts for you.

If you prefer to align them further, I can apply the refactoring 👍

Reply:
No worries, please go ahead 👍 

I rather prefer an consistent API over some temporary merge conflicts 😄

Reply:
@stoerti commit [65f0a43](https://github.com/camunda/camunda/pull/25921/commits/65f0a437434e65b70a0b5a0e931b57692fb44516) renames the `GroupEntity#key` and `GroupItem#key` properties to `groupKey`.

Reply:
Nice, TY